cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/cpp/DeclarationHandler.kt

```diff
@@ -213,11 +215,12 @@ class DeclarationHandler(lang: CXXLanguageFrontend) :
 
         frontend.processAttributes(declaration, ctx)
 
-        frontend.scopeManager.leaveScope(declaration)
-        if (recordDeclaration != null && outsideOfRecord) {
-            frontend.scopeManager.leaveScope(recordDeclaration)
+        if (holder != null && outsideOfScope) {
```

Comment:
Why are we leaving the scope of the holder before the scope of the declaration? We enter them in order ``holder -> declaration` so we should leave them in the order `declaration -> holder` right? Maybe I am missing something. In this case it would not make a difference though, I think.

Reply:
You are correct. This was probably a problem by moving code fragments around.

Reply:
Reverted back to the previous correct way.